packages/containers/src/PieChartButton/PieChartButton.connect.js

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Immutable from 'immutable';
+import omit from 'lodash/omit';
+import { componentState, cmfConnect, Inject } from '@talend/react-cmf';
+import PieChartButton from '@talend/react-components/lib/PieChartButton';
+
+export const DEFAULT_STATE = new Immutable.Map({});
+
+export function ContainerPieChartButton(props) {
+	let overlayComponent = null;
+	let onClick = null;
+
+	if (typeof props.overlayComponent === 'string' && props.overlayComponent) {
+		overlayComponent = (
+			<Inject component={props.overlayComponent} {...props.overlayComponentProps} />
+		);
+	}
+
+	if (!props.onClick) {
+		onClick = (event, data) => {
+			if (props.actionCreator) {
+				props.dispatchActionCreator(props.actionCreator, event, data);
+			} else {
+				props.dispatch({
+					model: props.model,
+					...props.payload,
+				});
+			}
+		};
+	}
```

Comment:
Those tow `let` and initialization should be turned into functions. Those will be testable.